### PR TITLE
Scalajs

### PR DESCRIPTION
**Pull request by kkasravi**
Merge of #952

### CODE REVIEW

**Comment by kkasravi** [comment]
@stanleyxu2005 Stanley can you review (+1|-1) and I'll merge?.

**Comment by kkasravi** [comment]
Will need to redo PR based on #952 and merged master

File: core/src/main/scala/org/apache/gearpump/metrics/AkkaReporter.scala [hunk 1]
```diff
@@ -38,25 +40,22 @@ class AkkaReporter(system: ActorSystem, registry: MetricRegistry, marker: Marker
 
       val sgauges = collection.SortedMap(gauges.toSeq: _*)
       sgauges.foreach(pair => {
-        import org.apache.gearpump.metrics.Metrics._
         val (key, value) = pair
-        system.eventStream.publish(Gauge(key, value))
+        system.eventStream.publish(_Guage(key, value))
```
Comment (stanleyxu2005) [inline core/src/main/scala/org/apache/gearpump/metrics/AkkaReporter.scala]: Type -> _Gauge

File: streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/StreamingAppMasterDetail.scala [hunk 2]
```diff
@@ -34,7 +34,7 @@ case class StreamingAppMasterDataDetail(
     processorLevels: Map[ProcessorId, Int],
     dag: Graph[ProcessorId, PartitionerDescription] = null,
     actorPath: String = null,
-    clock: TimeStamp = 0,
+    clock: String = null,
```
Comment (stanleyxu2005) [inline streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/StreamingAppMasterDetail.scala]: Why the type is changed to string?

Comment (kkasravi) [inline streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/StreamingAppMasterDetail.scala]: Long ends up being opaque in scala-jus

**Pull request closed without merging**